### PR TITLE
spaces kind of worked in a hackey way

### PR DESCRIPTION
**Pull request by brickstar**
I should have abandoned the hash idea much earlier and started over, instead I spent most of my time trying to get my idea to work, when it wasn't the most expeditious solution

### CODE REVIEW

File: lib/caesar.rb [hunk 1]
```diff
@@ -1,6 +1,44 @@
 class Caesar
```
Comment (BrianZanti) [inline lib/caesar.rb:1]: Good git workflow!

File: test/caesar_test.rb [hunk 2]
```diff
@@ -1,2 +1,37 @@
...
+
+  def test_it_exists
+    cc = Caesar.new
+
+    assert_instance_of Caesar, cc
+  end
+
+  def test_one_letter
+    cc = Caesar.new
+    cc.cypher(3)
```
Comment (BrianZanti) [inline test/caesar_test.rb:17]: Because you have a method "cypher", you should have a test for it.

File: test/caesar_test.rb [hunk 3]
```diff
@@ -1,2 +1,37 @@
...
+
+class CaesarTest < Minitest::Test
+
+  def test_it_exists
+    cc = Caesar.new
+
+    assert_instance_of Caesar, cc
+  end
+
+  def test_one_letter
```
Comment (BrianZanti) [inline test/caesar_test.rb:15]: Really like that you're starting with the simple case and moving to the complex

File: test/caesar_test.rb [hunk 4]
```diff
@@ -1,2 +1,37 @@
...
+    cc = Caesar.new
+    cc.cypher(3)
+    assert_equal "q", cc.one_letter("t")
+  end
+
+  def test_encrypt
+    cc = Caesar.new
+    assert_equal "qeb nrfzh yoltk clu grjmp lsbo qeb ixwv ald", cc.eng_to_cypher("the quick brown fox jumps over the lazy dog", 3)
+  end
+
```
Comment (BrianZanti) [inline test/caesar_test.rb:25]: Missing test for case sensitivity

File: lib/caesar.rb [hunk 5]
```diff
@@ -1,6 +1,44 @@
...
+
+  def from_file(file_location, rotation)
+    File.open(file_location).each do |line|
+      line = line.strip
+      @phrase = line.to_s
+    end
+    eng_to_cypher(@phrase, rotation)
+  end
+
+  def eng_to_cypher(phrase, rotation)
```
Comment (BrianZanti) [inline lib/caesar.rb:30]: This should be eng_to_cipher

File: lib/caesar.rb [hunk 6]
```diff
@@ -1,6 +1,44 @@
...
+    cypher(rotation)
+    phrase_array = phrase.chars
+    phrase_array.map do |char|
+      one_letter(char)
+    end.join
+  end
```
Comment (BrianZanti) [inline lib/caesar.rb:36]: This method is very readable and short.

File: lib/caesar.rb [hunk 7]
```diff
@@ -1,6 +1,44 @@
 class Caesar
-  attr_reader :key
-
+  attr_reader :character_hash
   def initialize
+    @characters = ("a".."z")
+    @character_hash = {}
+  end
+
+  def rotate_characters(rotation)
```
Comment (BrianZanti) [inline lib/caesar.rb:8]: Good use of helper methods

File: lib/caesar.rb [hunk 8]
```diff
@@ -1,6 +1,44 @@
...
+    @characters.to_a.rotate(rotation)
+  end
+
+  def cypher(rotation)
```
Comment (BrianZanti) [inline lib/caesar.rb:12]: I would have named this a little bit more specifically. For instance, create_cipher, or generate_cipher

File: lib/caesar.rb [hunk 9]
```diff
@@ -1,6 +1,44 @@
...
+    rotated = rotate_characters(- rotation)
+    @character_hash = Hash[@characters.zip(rotated)]
+    @character_hash[" "] = " "
```
Comment (BrianZanti) [inline lib/caesar.rb:15]: You say it's hacky, but I love what you did with the Hash! Adding the space to the hash is a very simple way to get around that character. I can see your logic here... we are mapping characters to their encodings, and that's what a hash does. It maps one key to a value. One of the cooler ways I've seen to do this.

File: lib/caesar.rb [hunk 10]
```diff
@@ -1,6 +1,44 @@
 class Caesar
-  attr_reader :key
-
+  attr_reader :character_hash
   def initialize
+    @characters = ("a".."z")
```
Comment (BrianZanti) [inline lib/caesar.rb:4]: You were so close to getting case sensitivity! You would have just needed to do the same thing with the hash with the upper case characters.

File: lib/caesar.rb (hunk 9 above)
Comment (BrianZanti) [inline lib/caesar.rb:15]: Think about how we can make this work for any nonalphabetic character

Comment (brickstar) [reply lib/caesar.rb:15]: Thanks for all the comments and insight, really appreciate you spending the time to provide thorough feedback.

**Pull request closed without merging**